include/deal.II/base/tensor.h

```diff
@@ -300,153 +558,384 @@ class Tensor
+}
+
+
+template <int dim, typename Number>
+inline
+Tensor<0,dim,Number> Tensor<0,dim,Number>::operator - () const
+{
+  return -value;
+}
+
+
+template <int dim, typename Number>
+inline
+typename Tensor<0,dim,Number>::real_type
+Tensor<0,dim,Number>::norm () const
+{
+  Assert(dim != 0, ExcMessage("Cannot access an object of type Tensor<0,0,Number>"));
+  return numbers::NumberTraits<Number>::abs (value);
+}
+
+
+template <int dim, typename Number>
+inline
+typename Tensor<0,dim,Number>::real_type
+Tensor<0,dim,Number>::norm_square () const
+{
+  Assert(dim != 0, ExcMessage("Cannot access an object of type Tensor<0,0,Number>"));
+  return numbers::NumberTraits<Number>::abs_square (value);
+}
+
+
+template <int dim, typename Number>
+template <typename OtherNumber>
+inline
+void
+Tensor<0, dim, Number>::unroll_recursion (Vector<OtherNumber> &result,
+                                          unsigned int        &index) const
+{
+  Assert(dim != 0, ExcMessage("Cannot unroll an object of type Tensor<0,0,Number>"));
+  result[index] = value;
+  ++index;
+}
+
+
+template <int dim, typename Number>
+inline
+void Tensor<0,dim,Number>::clear ()
+{
+  value = value_type();
+}
+
+
+template <int dim, typename Number>
+template <class Archive>
+inline
+void Tensor<0,dim,Number>::serialize(Archive &ar, const unsigned int)
+{
+  ar &value;
+}
+
+
+/*-------------------- Inline functions: Tensor<rank,dim> --------------------*/
+
+
+namespace internal
+{
+  // TODO: Think about refactoring this into the TableIndices class as a
+  // general, polymorphic for extracting an item out of an object with
+  // nested identifiers.
+  template<int rank_> struct TensorIndicesHelper
+  {
+    // used for implementing Tensor<rank,dim>::operator[] with TableIndices
+    // tail recursive call to form up access to
+    //   tensor[indices[0]][indices[1]]...[indices[rank_]]
+    template<int rank, int dim, typename Number>
+    static inline
+    Number &extract(Tensor<rank_,dim,Number> &t, const TableIndices<rank> &indices)
+    {
+      Assert (indices[rank - rank_]<dim, ExcIndexRange (indices[rank - rank_], 0, dim));
+      return TensorIndicesHelper<rank_ - 1>::template extract<rank, dim, Number>(
+        t[indices[rank - rank_]], indices);
+    }
+  };
+
+  template<> struct TensorIndicesHelper<1>
+  {
+    template<int rank, int dim, typename Number>
+    static inline
+    Number &extract(Tensor<1,dim,Number> &t, const TableIndices<rank> &indices)
+    {
+      Assert (indices[rank - 1]<dim, ExcIndexRange (indices[rank - 1], 0, dim));
+      return t[indices[rank-1]];
+    }
+  };
+} /* internal */
+
+
+template <int rank_, int dim, typename Number>
+inline
+Tensor<rank_,dim,Number>::Tensor (const bool /*initialize*/)
+{
+  // All members of the c-style array values are already default initialized
+  // and thus all values are already set to zero recursively.
+
+  // TODO: Think about using the default heap allocator to restore the old
+  // behavior.
+}
```

Comment:
I will remove it. I doubt that this boolean is used terribly often in user code.